Backpack/src/main/java/net/skyscanner/backpack/calendar/view/BpkCalendarLabelsViewModel.kt

```diff
@@ -50,38 +67,46 @@ internal class BpkCalendarLabelsViewModel(
       if (field != value) {
         field = value
         labels.forEach { it.value.width = cellWidth }
+        placeholder.width = cellWidth
       }
     }
 
   fun draw(canvas: Canvas, date: LocalDate, x: Float, y: Float, disabled: Boolean) {
-    if (disabled) {
+    if (disabled || isEmpty()) {
       return
     }
-    val label = labels[date]
-    if (label != null) {
-      label.draw(canvas, x, y)
-    } else if (labels.isNotEmpty()) {
-      canvas.drawText("-", x, y, paints.getValue(CalendarLabel.Style.PriceMedium))
-    }
+    val label = labels.getOrElse(date) { placeholder }
+    label.draw(canvas, x, y)
   }
 
   class BpkCalendarLabelViewModel(
     private val label: CalendarLabel,
     private val paint: TextPaint,
   ) {
 
-    private lateinit var ellipsizedText: String
+    private lateinit var text: StaticLayout
```

Comment:
StaticLayout is a class from Android library responsible for proper multi-line slicing. It also supports text truncating